osquery/core/conversions.h

```diff
@@ -322,4 +306,102 @@ inline typename std::enable_if<
+    throwingStringToIntImpl(const FromType& from, const int base) {
+  auto pos = std::size_t{};
+  return std::stol(from, &pos, base);
+}
+
+template <typename ToType, typename FromType>
+inline
+    typename std::enable_if<std::is_same<ToType, long long int>::value &&
+                                (std::is_same<FromType, std::string>::value ||
+                                 std::is_same<FromType, std::wstring>::value),
+                            ToType>::type
+    throwingStringToIntImpl(const FromType& from, const int base) {
+  auto pos = std::size_t{};
+  return std::stoll(from, &pos, base);
+}
+
+template <typename ToType, typename FromType>
+inline
+    typename std::enable_if<std::is_same<ToType, unsigned int>::value &&
+                                (std::is_same<FromType, std::string>::value ||
+                                 std::is_same<FromType, std::wstring>::value),
+                            ToType>::type
+    throwingStringToIntImpl(const FromType& from, const int base) {
+  auto pos = std::size_t{};
+  return std::stoul(from, &pos, base);
+}
+
+template <typename ToType, typename FromType>
+inline
+    typename std::enable_if<std::is_same<ToType, unsigned long int>::value &&
+                                (std::is_same<FromType, std::string>::value ||
+                                 std::is_same<FromType, std::wstring>::value),
+                            ToType>::type
+    throwingStringToIntImpl(const FromType& from, const int base) {
+  auto pos = std::size_t{};
+  return std::stoul(from, &pos, base);
+}
+
+template <typename ToType, typename FromType>
+inline typename std::enable_if<
+    std::is_same<ToType, unsigned long long int>::value &&
+        (std::is_same<FromType, std::string>::value ||
+         std::is_same<FromType, std::wstring>::value),
+    ToType>::type
+throwingStringToIntImpl(const FromType& from, const int base) {
+  auto pos = std::size_t{};
+  return std::stoull(from, &pos, base);
+}
+
+template <typename ToType, typename FromType>
+inline
+    typename std::enable_if<std::is_integral<ToType>::value &&
+                                !std::is_same<ToType, bool>::value &&
+                                (std::is_same<FromType, std::string>::value ||
+                                 std::is_same<FromType, std::wstring>::value),
+                            Expected<ToType, ConversionError>>::type
+    tryTo(const FromType& from, const int base = 10) noexcept {
+  try {
+    return throwingStringToIntImpl<ToType>(from, base);
+  } catch (const std::invalid_argument& ia) {
+    return createError(ConversionError::InvalidArgument,
+                       "If no conversion could be performed. ")
+           << ia.what();
+  } catch (const std::out_of_range& oor) {
+    return createError(ConversionError::OutOfRange,
+                       "Value read is out of the range of representable values "
+                       "by an int. ")
+           << oor.what();
+  }
```

Comment:
`noexcept` is a pretty strong guarantee - `std::terminate` may be called if this is violated if I remember correctly. What do you think about handling unknown exceptions as well?

Reply:
I'm not sure about it.
The `osquery` itself doesn't use exceptions that means any exceptions likely leads to crash. Thats a good reason to mark functions as `noexcept`.

You're right, any exceptions in case of `noexcept` lead to `terminate`. But there is only one function call inside: `std::sto(i,l,ll,u,ul)`, which according to [API](https://en.cppreference.com/w/cpp/string/basic_string/stol) throws only two types of exception. Do you think it could be some other exception?

Reply:
Ok, may be it is a good reason. I've added _catch-all_ handler. Thank you for the help with it!